gravitee-am-service/src/main/java/io/gravitee/am/service/impl/PasswordPolicyServiceImpl.java

```diff
@@ -64,4 +72,26 @@ public Single<PasswordPolicy> create(ReferenceType referenceType, String referen
                         .principal(principal)
                         .type(EventType.PASSWORD_POLICY_CREATED).throwable(error)));
     }
+
+    @Override
+    public Maybe<PasswordPolicy> retrievePasswordPolicy(io.gravitee.am.model.User user, PasswordSettingsAware passwordSettingsAware) {
+        // TODO this implementation will change once IDP will contains the policyId
+        //      as the rule will be to provide the Policy linked to the user.getSource IDP
+        //      and if no policy if found fallback to this implementation where app PasswordSettings
+        //      will have precedence on the default policy linked to the domain
+        return passwordSettingsAware == null ? defaultPasswordPolicy(user) : Optional.ofNullable(passwordSettingsAware)
```

Comment:
there is check that object is not null

```suggestion
        return passwordSettingsAware == null ? defaultPasswordPolicy(user) : Optional.of(passwordSettingsAware)
```